src/Kernel32.Desktop/Kernel32.cs

```diff
@@ -410,6 +409,172 @@ public enum SecurityImpersonationLevel
+        /// function to retrieve the full name of an executable image for both 32- and 64-bit processes from a 32-bit process.
+        /// </para>
+        /// </remarks>
+        [DllImport(nameof(Kernel32), SetLastError = true)]
+        public static extern SafeObjectHandle CreateToolhelp32Snapshot(
+            CreateToolhelp32SnapshotFlags dwFlags,
+            uint th32ProcessID);
+
+        /// <summary>Retrieves information about the first process encountered in a system snapshot.</summary>
+        /// <param name="hSnapshot">
+        /// A handle to the snapshot returned from a previous call to the
+        /// <see cref="CreateToolhelp32Snapshot" /> function.
+        /// </param>
+        /// <param name="lppe">
+        /// Contains process information such as the name of the executable file, the process identifier, and
+        /// the process identifier of the parent process.
+        /// </param>
+        /// <returns>
+        /// Returns <see langword="true" /> if the first entry of the process list has been copied to the buffer or
+        /// <see langword="false" /> otherwise. The <see cref="Win32ErrorCode.ERROR_NO_MORE_FILES" /> error value is returned by
+        /// the <see cref="Marshal.GetLastWin32Error" /> function if no processes exist or the snapshot does not contain process
+        /// information.
+        /// </returns>
+        [DllImport(nameof(Kernel32), SetLastError = true, CharSet = CharSet.Unicode)]
+        public static extern bool Process32First(SafeObjectHandle hSnapshot, [In, Out] PROCESSENTRY32 lppe);
+
+        /// <summary>Retrieves information about the next process recorded in a system snapshot.</summary>
+        /// <param name="hSnapshot">
+        /// A handle to the snapshot returned from a previous call to the
+        /// <see cref="CreateToolhelp32Snapshot" /> function.
+        /// </param>
+        /// <param name="lppe">A <see cref="PROCESSENTRY32" /> structure.</param>
+        /// <returns>
+        /// Returns <see langword="true" /> if the next entry of the process list has been copied to the buffer or
+        /// <see langword="false" /> otherwise. The <see cref="Win32ErrorCode.ERROR_NO_MORE_FILES" /> error value is returned by
+        /// the <see cref="Marshal.GetLastWin32Error" /> function if no processes exist or the snapshot does not contain process
+        /// information.
+        /// </returns>
+        /// <remarks>
+        /// To retrieve information about the first process recorded in a snapshot, use the
+        /// <see cref="Process32First(SafeObjectHandle,PROCESSENTRY32)" />
+        /// function.
+        /// </remarks>
+        [DllImport(nameof(Kernel32), SetLastError = true, CharSet = CharSet.Unicode)]
+        public static extern bool Process32Next(
+            SafeObjectHandle hSnapshot,
+            [In, Out] PROCESSENTRY32 lppe);
```

Comment:
It's actually due to the `StructLayout` on classes thing. And i'll need to go through the other place i've used it and check because I might need to add `[In, Out]` there too.

The problem is that by default `FOO_CLASS foo` is marshalled to the lib but not marshalled back. The solution with structures is to add `ref`but it doesn't work with classes as `ref FOO_CLASS foo` seem to be marshalled as `FOO_CLASS**` (I doesn't work so that's what I suspect and it would be logic) but that's not what we want so `[Out]` is added. But adding `[Out]` actually remove the automatic behavior of `[In]` so I must add it back too to get the wanted behavior.

I'm very happy to have learned about putting `StructLayout` on classes, it produce really clean code but there is clearly a few things to take in account while using it, maybe we should write a readme somewhere if we agree that it's the way to go.

Also should we convert existing structs ? The out/ref syntax is more common for PInvoke but standardizing on a single way to do that across all the lib would make it predictable.